### PR TITLE
Add truncate functionality

### PR DESCRIPTION
**Pull request by patrickdronk**
I've added the truncate functionality as described [here](https://github.com/Ff00ff/mammoth/issues/132)
Things that should be done after merge, which I'll do as well:
- Update docs
- Checks? I've seen this directory. But not sure what the purpose of it is. Is there something I can read to get some context?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -267,6 +267,13 @@ Below is a list of clauses per query and a short description on what we Mammoth
 - [ RETURNING \* | output_expression [ [ AS ] output_name ] [, ...] ] — supported, but limited to 10 expressions
 </details>
 
+<details>
+  <summary>Trunate</summary>
```
Comment (martijndeh) [inline README.md:271]: Eeks minor typo

File: src/truncate.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+export class TruncateQuery<T extends Table<any, any>> extends Query<void> {
+    constructor(
+        private readonly queryExecutor: QueryExecutorFn,
+        private readonly table: T,
+        private readonly tokens: Token[],
+    ) {
+        super();
+    }
+
+    restartIdentity<T extends Table<any, any>>() {
```
Comment (martijndeh) [inline src/truncate.ts:26]: Nice-to-have to also add support for `continueIdentity()` and `restrict()`.

**Review by martijndeh — COMMENTED** [review]
Great, you pretty much nailed it! There are some minor clean ups possible but that's fine. Are you running prettier? I should probably add a npm script for this.

As for the checks yes that's a good point which I should explain a bit better. The checks are basically type-checking unit tests and makes sure we don't create any regressions with the types. With all the advanced typings going on it's very easy to make a mistake and get an `any` somewhere. Regular runtime unit tests don't really cover this so we use dts-jest. https://www.npmjs.com/package/dts-jest

It would be nice to get at least one check in to see the return type of the truncate query. I'm assuming this should be a number. The delete check is actually very similar (and simple), so you could try and copy that one. What do you think?

It looks like the tests etc aren't running on this branch yet so I have some work to do. I'll get this sorted in the next couple of days.

Good stuff!

**Comment by patrickdronk** [comment]
Hi @martijndeh 
Thanks for your feedback, I've added the prettier package now, styling should be more in line with the project now.
I've read up on the type testing. I've basically used the delete, and changed it ^^. I hope it what you would expect from it. 
Let me know what you think of the last changes 👋

**Comment by martijndeh** [comment]
I fixed the tests and everything was green so this is in now! Nice work!

**Pull request merged**